### PR TITLE
[Improvement](segment iterator) Optimize column row reservation to reduce overhead

### PR DESCRIPTION
**Pull request by airborne12**
## Proposed changes

This PR improves the segment iterator by reducing the reserved rows for a column when the row bitmap is smaller than the block row. This optimization aims to enhance memory efficiency and improve performance by reducing unnecessary row allocations.

### CODE REVIEW

**Comment by airborne12** [comment]
run buildall

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp [hunk 1]
```diff
@@ -1989,8 +1989,11 @@ Status SegmentIterator::_next_batch_internal(vectorized::Block* block) {
     if (UNLIKELY(!_lazy_inited)) {
         RETURN_IF_ERROR(_lazy_init());
         _lazy_inited = true;
+        uint32_t nrows_reserve_limit = _row_bitmap.cardinality() < _opts.block_row_max
+                                               ? _row_bitmap.cardinality()
```
Comment (eldenmoon) [inline be/src/olap/rowset/segment_v2/segment_iterator.cpp:1993]: add comment?
std::min(_row_bitmap.cardinality(), _opts.block_row_max) is better

**Comment by airborne12** [comment]
run buildall

**Comment by airborne12** [comment]
run buildall

**Review by zzzxl1993 — APPROVED** [review]
LGTM

**Review by qidaye — APPROVED** [review]
LGTM

**Comment by airborne12** [comment]
run p1

**Comment by airborne12** [comment]
run cloud_p1

**Pull request merged**